server_tests/test/clientTests.js

```diff
@@ -791,514 +736,468 @@ describe("RUST API TEST SUITE", function() {
+            .then( shouldNotGetHere, function onError (err) {
+                Comparison.invalidParameterValueError(err);
+            });
         });
     });
 
     // BUBL POLL
     describe.skip("Testing /osc/commands/_bublPoll endpoint", function() {
         var sessionId;
 
-        before( function(done) {
+        before( function() {
             if (!isBublcam) {
               return this.skip();
             }
 
-            testClient.startSession()
-            .then( Comparison.catchExceptions(done, function(res) {
+            return testClient.startSession()
+            .then( function onSuccess (res) {
                 sessionId = res.body.results.sessionId;
                 Comparison.oscSessionOpOutput(res, {'sessionId': sessionId});
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        after( function(done) {
-            testClient.closeSession(sessionId)
-            .then( Comparison.catchExceptions(done, function(res) {
+        after( function() {
+            return testClient.closeSession(sessionId)
+            .then( function onSuccess (res) {
                 Comparison.oscCloseSessionOutput(res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        it("Expect success. /osc/commands/_bublPoll returns immediately if no waitTimeout argument is provided", function(done) {
+        it("Expect success. /osc/commands/_bublPoll returns immediately if no waitTimeout argument is provided", function() {
             this.timeout(timeoutValue);
             var fingerprint = '';
-            testClient.takePicture(sessionId, function(res) {
+            return testClient.takePicture(sessionId, function(res) {
                 var commandId = res.body.id;
-                testClient.bublPoll(res.body.id, fingerprint)
-                .then( Comparison.catchExceptions(done, function(res) {
+                return testClient.bublPoll(res.body.id, fingerprint)
+                .then( function onSuccess (res) {
                     Comparison.bublPollOutput(res, true, {'id': commandId, 'fingerprint': fingerprint});
-                }));
+                }, shouldNotGetHere);
             })
-            .then( Comparison.catchExceptions(done, function(res) {
+            .then( function onSuccess (res) {
                 Comparison.oscTakePictureOutput(res);
-                done();
-            }));
+            }, shouldNotGetHere);
         });
 
-        it("Expect success. /osc/commands/_bublPoll returns once command state has changed", function(done) {
+        it("Expect success. /osc/commands/_bublPoll returns once command state has changed", function() {
             this.timeout(timeoutValue);
             var fingerprint = '';
             var commandId = '';
-            testClient.bublCaptureVideo(sessionId, function(res) {
+            return testClient.bublCaptureVideo(sessionId, function(res) {
```

Comment:
This test probably needs to use deferred promises, because it spawns new tests in the status callback.